### ISSUE TITLE
etcdserver advertise url does not match the url which specified from cli

### ISSUE THREAD

**Issue by gouyang** [coreos/etcd#2694]
Run the etcd from command line with IP address in peer url and client url, etcdserver advertise still use "localhost".

```bash
$ ./bin/etcd --listen-peer-urls="http://10.66.11.39:2380,http://10.66.11.39:7001" --listen-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001"
2015/04/16 15:59:30 etcd: no data-dir provided, using default data-dir ./default.etcd
2015/04/16 15:59:30 etcd: already initialized as member before, starting as etcd member...
2015/04/16 15:59:30 etcd: listening for peers on http://10.66.11.39:2380
2015/04/16 15:59:30 etcd: listening for peers on http://10.66.11.39:7001
2015/04/16 15:59:30 etcd: listening for client requests on http://10.66.11.39:2379
2015/04/16 15:59:30 etcd: listening for client requests on http://10.66.11.39:4001
2015/04/16 15:59:30 etcdserver: datadir is valid for the 2.0.1 format
2015/04/16 15:59:30 etcdserver: name = default
2015/04/16 15:59:30 etcdserver: data dir = default.etcd
2015/04/16 15:59:30 etcdserver: member dir = default.etcd/member
2015/04/16 15:59:30 etcdserver: heartbeat = 100ms
2015/04/16 15:59:30 etcdserver: election = 1000ms
2015/04/16 15:59:30 etcdserver: snapshot count = 10000
2015/04/16 15:59:30 etcdserver: advertise client URLs = http://localhost:2379,http://localhost:4001
2015/04/16 15:59:30 etcdserver: restart member ce2a822cea30bfca in cluster 7e27652122e8b2ae at commit index 5020
2015/04/16 15:59:30 INFO: raft: ce2a822cea30bfca became follower at term 2
2015/04/16 15:59:30 INFO: raft: newRaft ce2a822cea30bfca [peers: [], term: 2, commit: 5020, applied: 0, lastindex: 5020, lastterm: 2]
2015/04/16 15:59:30 etcdserver: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
2015/04/16 15:59:31 INFO: raft: ce2a822cea30bfca is starting a new election at term 2
2015/04/16 15:59:31 INFO: raft: ce2a822cea30bfca became candidate at term 3
2015/04/16 15:59:31 INFO: raft: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 3
2015/04/16 15:59:31 INFO: raft: ce2a822cea30bfca became leader at term 3
2015/04/16 15:59:31 INFO: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 3
2015/04/16 15:59:31 etcdserver: published {Name:default ClientURLs:[http://localhost:2379 http://localhost:4001]} to cluster 7e27652122e8b2ae

$ ./bin/etcd -version
etcd version 2.0.4+git
```

**Comment by xiang90** [comment]
@gouyang You need to specify the `--advertise-url` for both peer and client url.

Please see https://github.com/coreos/etcd/blob/master/Documentation/configuration.md#-initial-advertise-peer-urls

**Comment by gouyang** [comment]
There is still one "localhost" exists in "added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae", is it right?


```
$ ./bin/etcd --listen-peer-urls="http://10.66.11.39:2380,http://10.66.11.39:7001" --listen-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001" --advertise-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001" --initial-advertise-peer-urls="http://10.66.11.39:2380,http://10.66.11.39:7001"
2015/04/17 10:17:34 etcd: no data-dir provided, using default data-dir ./default.etcd
2015/04/17 10:17:34 etcd: already initialized as member before, starting as etcd member...
2015/04/17 10:17:34 etcd: listening for peers on http://10.66.11.39:2380
2015/04/17 10:17:34 etcd: listening for peers on http://10.66.11.39:7001
2015/04/17 10:17:34 etcd: listening for client requests on http://10.66.11.39:2379
2015/04/17 10:17:34 etcd: listening for client requests on http://10.66.11.39:4001
2015/04/17 10:17:34 etcdserver: datadir is valid for the 2.0.1 format
2015/04/17 10:17:34 etcdserver: name = default
2015/04/17 10:17:34 etcdserver: data dir = default.etcd
2015/04/17 10:17:34 etcdserver: member dir = default.etcd/member
2015/04/17 10:17:34 etcdserver: heartbeat = 100ms
2015/04/17 10:17:34 etcdserver: election = 1000ms
2015/04/17 10:17:34 etcdserver: snapshot count = 10000
2015/04/17 10:17:34 etcdserver: advertise client URLs = http://10.66.11.39:2379,http://10.66.11.39:4001
2015/04/17 10:17:34 etcdserver: restart member ce2a822cea30bfca in cluster 7e27652122e8b2ae at commit index 5207
2015/04/17 10:17:34 INFO: raft: ce2a822cea30bfca became follower at term 5
2015/04/17 10:17:34 INFO: raft: newRaft ce2a822cea30bfca [peers: [], term: 5, commit: 5207, applied: 0, lastindex: 5207, lastterm: 5]
2015/04/17 10:17:34 etcdserver: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
2015/04/17 10:17:35 INFO: raft: ce2a822cea30bfca is starting a new election at term 5
2015/04/17 10:17:35 INFO: raft: ce2a822cea30bfca became candidate at term 6
2015/04/17 10:17:35 INFO: raft: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 6
2015/04/17 10:17:35 INFO: raft: ce2a822cea30bfca became leader at term 6
2015/04/17 10:17:35 INFO: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 6
2015/04/17 10:17:35 etcdserver: published {Name:default ClientURLs:[http://10.66.11.39:2379 http://10.66.11.39:4001]} to cluster 7e27652122e8b2ae
```

**Comment by xiang90** [comment]
@gouyang 

```
2015/04/17 10:17:34 etcd: already initialized as member before, starting as etcd member...
```

You have to remove your previous data-dir to get rid of the initial incorrect configuration.

**Comment by gouyang** [comment]
@xiang90 
use a new data-dir get same result, so no related to data-dir.

```
$ ./bin/etcd --name test --data-dir tmp --listen-peer-urls="http://10.66.11.39:2380,http://10.66.11.39:7001" --listen-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001" --advertise-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001"
2015/04/17 14:18:03 etcd: listening for peers on http://10.66.11.39:2380
2015/04/17 14:18:03 etcd: listening for peers on http://10.66.11.39:7001
2015/04/17 14:18:03 etcd: listening for client requests on http://10.66.11.39:2379
2015/04/17 14:18:03 etcd: listening for client requests on http://10.66.11.39:4001
2015/04/17 14:18:03 etcdserver: datadir is valid for the 2.0.1 format
2015/04/17 14:18:03 etcdserver: name = test
2015/04/17 14:18:03 etcdserver: data dir = tmp
2015/04/17 14:18:03 etcdserver: member dir = tmp/member
2015/04/17 14:18:03 etcdserver: heartbeat = 100ms
2015/04/17 14:18:03 etcdserver: election = 1000ms
2015/04/17 14:18:03 etcdserver: snapshot count = 10000
2015/04/17 14:18:03 etcdserver: advertise client URLs = http://10.66.11.39:2379,http://10.66.11.39:4001
2015/04/17 14:18:03 etcdserver: initial advertise peer URLs = http://localhost:2380,http://localhost:7001
2015/04/17 14:18:03 etcdserver: initial cluster = test=http://localhost:2380,test=http://localhost:7001
2015/04/17 14:18:03 etcdserver: start member ce2a822cea30bfca in cluster 7e27652122e8b2ae
2015/04/17 14:18:03 INFO: raft: ce2a822cea30bfca became follower at term 0
2015/04/17 14:18:03 INFO: raft: newRaft ce2a822cea30bfca [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2015/04/17 14:18:03 INFO: raft: ce2a822cea30bfca became follower at term 1
2015/04/17 14:18:03 etcdserver: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
2015/04/17 14:18:05 INFO: raft: ce2a822cea30bfca is starting a new election at term 1
2015/04/17 14:18:05 INFO: raft: ce2a822cea30bfca became candidate at term 2
2015/04/17 14:18:05 INFO: raft: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 2
2015/04/17 14:18:05 INFO: raft: ce2a822cea30bfca became leader at term 2
2015/04/17 14:18:05 INFO: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 2
2015/04/17 14:18:05 etcdserver: published {Name:test ClientURLs:[http://10.66.11.39:2379 http://10.66.11.39:4001]} to cluster 7e27652122e8b2ae
```

**Comment by gouyang** [comment]
I finally get a correct version for all things, 

```
$ ./bin/etcd --name gh --listen-peer-urls="http://10.66.11.39:2380,http://10.66.11.39:7001" --listen-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001" --advertise-client-urls="http://10.66.11.39:2379,http://10.66.11.39:4001" --initial-advertise-peer-urls http://10.66.11.39:2380,http://10.66.11.39:7001 -initial-cluster gh=http://10.66.11.39:2380,gh=http://10.66.11.39:7001
2015/04/17 14:45:44 etcd: no data-dir provided, using default data-dir ./gh.etcd
2015/04/17 14:45:44 etcd: already initialized as member before, starting as etcd member...
2015/04/17 14:45:44 etcd: listening for peers on http://10.66.11.39:2380
2015/04/17 14:45:44 etcd: listening for peers on http://10.66.11.39:7001
2015/04/17 14:45:44 etcd: listening for client requests on http://10.66.11.39:2379
2015/04/17 14:45:44 etcd: listening for client requests on http://10.66.11.39:4001
2015/04/17 14:45:44 etcdserver: datadir is valid for the 2.0.1 format
2015/04/17 14:45:44 etcdserver: name = gh
2015/04/17 14:45:44 etcdserver: data dir = gh.etcd
2015/04/17 14:45:44 etcdserver: member dir = gh.etcd/member
2015/04/17 14:45:44 etcdserver: heartbeat = 100ms
2015/04/17 14:45:44 etcdserver: election = 1000ms
2015/04/17 14:45:44 etcdserver: snapshot count = 10000
2015/04/17 14:45:44 etcdserver: advertise client URLs = http://10.66.11.39:2379,http://10.66.11.39:4001
2015/04/17 14:45:44 etcdserver: restart member 86158c7e8602b7db in cluster d039d9f6ea97087c at commit index 122
2015/04/17 14:45:44 INFO: raft: 86158c7e8602b7db became follower at term 3
2015/04/17 14:45:44 INFO: raft: newRaft 86158c7e8602b7db [peers: [], term: 3, commit: 122, applied: 0, lastindex: 122, lastterm: 3]
2015/04/17 14:45:44 etcdserver: added local member 86158c7e8602b7db [http://10.66.11.39:2380] to cluster d039d9f6ea97087c
2015/04/17 14:45:45 INFO: raft: 86158c7e8602b7db is starting a new election at term 3
2015/04/17 14:45:45 INFO: raft: 86158c7e8602b7db became candidate at term 4
2015/04/17 14:45:45 INFO: raft: 86158c7e8602b7db received vote from 86158c7e8602b7db at term 4
2015/04/17 14:45:45 INFO: raft: 86158c7e8602b7db became leader at term 4
2015/04/17 14:45:45 INFO: raft.node: 86158c7e8602b7db elected leader 86158c7e8602b7db at term 4
2015/04/17 14:45:45 etcdserver: published {Name:gh ClientURLs:[http://10.66.11.39:2379 http://10.66.11.39:4001]} to cluster d039d9f6ea97087c
```
